rfc#0001

```diff
@@ -0,0 +1,140 @@
+that comprises it (binary/binaries + data elements) need to be loaded
+into it.  This is sent to it by the Enarx client agent (ECA),
+encrypted under a session key established at Keep provisioning time,
+and communicated to the ECA - the workload provisioning encryption
+key (WPEK).  The Keep has an agent running which is listening on a
+well-known port, or on a port number previously established during
+the Keep preparation phase.  The ECA connects to this port, and
+establishes a connection to the Keep.  The Keep takes the workload,
+decrypts it, and runs it.
+
+For this design to work, a route between the ECA and the Keep control
+plane needs to be available.  Note that this does not necessarily
+need to be direct: since an encrypted connection is not explicitly
+required (see below), proxying is possible.  This could, if required,
+be via an (untrusted) service on the host.
+
+One of the Enarx project's design principles is that all network
+and storage connections in and out of a Keep are encrypted.  In the
+case of network connections, this usually means that the session is
+encrypted using TLS (or dTLS), but since we may not have established
+any certificates in the Keep at this point, it cannot act as the
+server.  However, as the workload provisioning encryption key (WPEK)
+has already been communicated to the Keep, if the workload can be
+decrypted using this key, this assures the Keep that it is
+communicating with the ECA, or at least an entity in the same trust
+domain, so identity has been confirmed.
+
+There are possible use cases where a Keep might wish to cache a copy
+of the workload, either for reloading ("recycling" a Keep) or to
+transfer to another Keep in the same trust domain.  This caching, if
+under the origincal WPEK, is at the expense of Perfect Forward
+Secrecy.  If a Keep is to transfer the workload to another Keep in
+the same domain, it MAY encrypt it under a new WPEK.  If the
+original is used, it will need to established in the new Keep either
+at provisioning time (e.g. by the ECA) or at a later date through
+key negotiation.  Such functionality is considered risky, as the key
+management required is difficult to get right, and provides new
+options for compromise or design/implementation errors.  Any design
+of such functionality requires a new RFC (see "Future possibilities"
+below).
+
+The Keep MUST NOT accept unencrypted workloads, as there is no way
+to verify their provenance.
+The Keep SHOULD listen on a well-known port (see "Unresolved
+questions").
+If a port has been pre-esablished, the Keep MUST listen on this port,
+and MAY also listen on the well-known port.
+The Keep MAY cache the workload for re-use, or transfer to another
+Keep.  If a Keep allows workloads to be used by another Keep, the
+workload MAY be encrypted under a new WPEK, established between the
+original Keep and new Keep.
+Keeps MAY be implemented to enable the capability to performing
```

Comment:
I'm never sure what people mean by *implement*. Does this merely mean configured in this context?

Reply:
No, it means coded.  This should become a design document which allows different implementations to exist which are all compliant with this RFC, but which exhibit different behaviour.  Some might be newer versions, some might exist only on particular architectures, and others might be clean-room implementations.

Reply:
Thanks for elaborating. It shows me that I have some getting accustomed to do for the RFC language :-)

Reply:
@axelsimon I'll leave the resolving of this comment to you to acknowledge Mike's mention.

Reply:
Discussed with @axelsimon .